crates/sui-core/src/authority.rs

```diff
@@ -2092,6 +2102,9 @@ impl AuthorityState {
             .tap_ok(|_| {
                 debug!(?digest, ?effects_digest, ?self.name, "commit_certificate finished");
             })?;
+        self.metrics
+            .pending_notify_read
+            .set(self.database.notify_read.num_pending() as i64);
```

Comment:
This seems like an odd place to set this metric - can we not set it inside the NotifyRead impl?

Reply:
This is actually pretty hard to do, because we don't have any metrics in AuthorityStore and passing metrics there is going to be PR larger then what we have now.

I can either remove metric for now and introduce it later, or we can just keep this place even if it is not super obvious?